plugins/woocommerce/client/blocks/assets/js/base/context/hooks/cart/test/use-store-cart-coupons.tsx

```diff
@@ -0,0 +1,414 @@
+								needs_shipping: false,
+								shipping_address: {},
+								billing_address: {},
+							},
+						},
+					],
+				} )
+			);
+
+			const { result } = renderHook( () =>
+				useStoreCartCoupons( 'wc/checkout' )
+			);
+
+			// Apply a coupon
+			await act( async () => {
+				await result.current.applyCoupon( 'TEST5' );
+			} );
+
+			// Find the batch API call (filter out any initial cart fetches)
+			const batchCalls = fetchMock.mock.calls.filter(
+				( call ) =>
+					typeof call[ 0 ] === 'string' &&
+					call[ 0 ].includes( '/wc/store/v1/batch' )
+			);
+
+			// Verify exactly one batch API call was made
+			expect( batchCalls ).toHaveLength( 1 );
+			expect( batchCalls[ 0 ][ 0 ] ).toEqual(
+				expect.stringContaining( '/wc/store/v1/batch' )
+			);
+			expect( batchCalls[ 0 ][ 1 ] ).toEqual(
+				expect.objectContaining( {
+					method: 'POST',
+					headers: expect.objectContaining( {
+						'Content-Type': 'application/json',
+					} ),
+					body: expect.stringContaining( 'apply-coupon' ),
+				} )
+			);
+
+			// Verify the request body contains the coupon code
+			const requestBody = batchCalls[ 0 ][ 1 ].body;
+			const parsedBody = JSON.parse( requestBody );
+			expect( parsedBody.requests ).toHaveLength( 1 );
+			expect( parsedBody.requests[ 0 ].path ).toBe(
+				'/wc/store/v1/cart/apply-coupon'
+			);
+			expect( parsedBody.requests[ 0 ].method ).toBe( 'POST' );
+			expect( parsedBody.requests[ 0 ].data.code ).toBe( 'TEST5' );
+		} );
+
+		it( 'calls API with correct data for multiple coupon types', async () => {
+			const testCoupons = [
+				'5fixedcheckout', // Fixed cart discount
+				'50percoffcheckout', // Percentage discount
+				'10fixedproductcheckout', // Fixed product discount
+			];
+
+			// Mock successful responses for each coupon
+			fetchMock.mockResponses(
+				[
+					JSON.stringify( {
+						responses: [ { status: 200, body: {} } ],
+					} ),
+					{ status: 200 },
+				],
+				[
+					JSON.stringify( {
+						responses: [ { status: 200, body: {} } ],
+					} ),
+					{ status: 200 },
+				],
+				[
+					JSON.stringify( {
+						responses: [ { status: 200, body: {} } ],
+					} ),
+					{ status: 200 },
+				]
+			);
+
+			const { result } = renderHook( () =>
+				useStoreCartCoupons( 'wc/checkout' )
+			);
+
+			// Apply each coupon and verify API calls
+			for ( let i = 0; i < testCoupons.length; i++ ) {
+				await act( async () => {
+					await result.current.applyCoupon( testCoupons[ i ] );
+				} );
+
+				// Verify the API call for this coupon
+				const callIndex = i;
+				const requestBody = fetchMock.mock.calls[ callIndex ][ 1 ].body;
+				const parsedBody = JSON.parse( requestBody );
+
+				expect( parsedBody.requests[ 0 ].path ).toBe(
+					'/wc/store/v1/cart/apply-coupon'
+				);
+				expect( parsedBody.requests[ 0 ].method ).toBe( 'POST' );
+				expect( parsedBody.requests[ 0 ].data.code ).toBe(
+					testCoupons[ i ]
+				);
+			}
+
+			expect( fetchMock ).toHaveBeenCalledTimes( 3 );
```

Comment:
do you want to also test for `toHaveBeenCalledWith`?

Reply:
Added 👌🏼